### DISCUSSION TITLE
How to disable sigma mouse wheel events in v3?

### DISCUSSION THREAD

**Discussion by HectorHS** [jacomyal/sigma.js#1492] [Q&A] [answered]
I want to disable the mouse wheel zoom effect on sigma, and keep the default mouse wheel to scroll functionality. I used to be able to do something like mouseWheelEnabled: false on sigma's settings but I cannot see any similar setting on V3. I can set         minCameraRatio: 1 & maxCameraRatio: 1 to effectively disable zoom but the default mouse wheel to scroll functionality is still prevented. I can do something like:  
```
const renderer = new Sigma(graph, container, { allowInvalidContainer: true });

renderer.on("wheelStage", (event) => {
            event.event.sigmaDefaultPrevented = true;
});
```
which prevents sigma's zoom in functionality, but again scrolling is disabled.

Is there a simple way to prevent the zooming functionality and reinstate scrolling on wheelStage events?

**Comment by jacomyal** [comment, marked as answer]
Hi @HectorHS,

Good catch, it was not possible to handle this properly with sigma.
I just released sigma v3.0.0, with a fix for this. Basically, sigma stops preventing default mouse-wheel behavior in two different cases:
- the new `enableCameraZooming` setting is true (showcased [here](https://www.sigmajs.org/storybook/?path=/story/camera-control--story))
- the settings `minCameraRatio` or `maxCameraRatio` are set, and the user hits the zoom boundaries (I added those settings to [the nodes pan story](https://www.sigmajs.org/storybook/?path=/story/mouse-manipulations--story), so that it is testable somewhere)

**Reply by HectorHS** [reply]
Hi @jacomyal, 

Amazing, thank you for this! 

enableCameraZooming works great for my use case! I got it working as expected on my project but I couldn't see it working on the showcase you linked, maybe it's missing the latest version?

I was not very much concerned about the minCameraRatio & maxCameraRatio personally, so I might not be the best one to comment on that but it seems to me these settings will now result in a different behaviour for everyone who is using them which 1) doesn't feel better than before from a UX perspective (by scrolling up and down a page you accidentally zoom in and out of the graph that often just disappears), and 2) there doesn't seem to be a way to return to the previous behaviour. 

I'd recommend to either a) keep preventing default mouse-wheel behaviour irrespective of zoom boundaries if  minCameraRatio and maxCameraRatio are set, or b)  allow the previous functionality when a combination of settings is set, eg 
```
enableCameraZooming: true,
minCameraRatio: 1,
maxCameraRatio: 2 
```
by allowing zooming boundaries and preventing normal wheel events at all times withing the graph boundaries.

Just to be clear: just the implementation of enableCameraZooming works great for me, I am just worried of other people's implementations.

Many thanks for your work!

**Reply by jacomyal** [reply]
OK, I get your point. I agree that it's some kind of breaking change (though, it's between the beta and the v3 release, so I can live with that ^^), but I actually think it is better, UX-wise:

It means that, from now on, sigma only blocks the natural scroll flow of a webpage when it actually has something to do with the wheel event. It looks more like what happens when a user scrolls in a block that is itself in another scrollable block.

I hope to have a better separation between how sigma handles events flow, what the camera is allowed to do, and what users are allowed to do (in terms of interactions) once I tackle issue #1475.

**Reply by HectorHS** [reply]
Sounds good to me - I just didn't want to be the cause of breaking changes to minCameraRatio & maxCameraRatio, settings I am not even using myself. It sounds like this is moving towards where you want to go and I am more than happy to trust you with that! 

Once again, thanks!
